testing-guidance/workflow-tests/scripting/lua-scripting-tests.md

```diff
@@ -0,0 +1,293 @@
+| Add local properties in a script | <ol><li>Add the following lines to the top of the OnDeactivate() function: <ul><li>`Bye = "Bye! "`</li><li>`NumberOne = 1`</li><li>`NumberTwo = 2`</li><li>`NumbersAdded = NumberOne + NumberTwo`</li></ul></li><li>Modify the Debug.Log line so that it uses the local properties - for example: `Debug.Log(tostring(Bye .. NumberOne .. " + " .. NumberTwo .. " = " .. NumbersAdded));`</li><li>Save the file and run and exit Game Mode with this script added to a Lua Script entity</li></ol> | <ul><li>File is saved without error</li><li>When exiting Game Mode, the properties are properly printed (for example *Bye! 1 + 2 = 3*)</li></ul> |
+
+**"Area: Properties usage in Lua scripts" script:**
+```
+-- BaseScript.lua 
+
+local BaseScript = 
+{
+    Properties =
+    {
+        MyValue = 1,
+    }
+}
+
+function BaseScript:OnActivate()
+     Debug.Log(tostring("Hi! My value is: " .. BaseScript.Properties.MyValue));
+end
+
+function BaseScript:OnDeactivate()
+     Bye = "Bye! "
+     NumberOne = 1
+     NumberTwo = 2
+     NumbersAdded = NumberOne + NumberTwo
+     Debug.Log(tostring(Bye .. NumberOne .. " + " .. NumberTwo .. " = " .. NumbersAdded));
+end
+
+return BaseScript
+```
+
+## Area: Debugging existing script
+
+### Project Requirements
+Any project with the Remote Tools Connection Gem added can be used and the Lua Editor has to be connected to Editor. Saved script from *Area: Properties usage in Lua scripts* helps with setup.
+
+**Product:** 
+Simple debugging setup created and confirmed to be working.
+
+**Suggested Time Box:**
+20 minutes
+
+| Workflow | Requests | Things to Watch For |
+|----------|----------|---------------------|
+| Open Debugging tools | <ol><li>Open Breakpoints tool using any method (Toolbar Breakpoints button, or *View → Breakpoints*) </li><li>Add breakpoints to some lines on your script using any method (Select a line and click either Toggle Breakpoint button on toolbar, *Debug → Toggle Breakpoint* or press *F9*. Alternatively left-click the area next to the line number on the script) </li><li>Remove added breakpoints (This can be done using the same methods as adding them)</li><li>Open Stack tool using any method (Toolbar Stack button or *View → Stack*) </li><li>Open LUA Locals tool using any method (Toolbar LUA Locals button or *View → LUA Locals*)</li></ol> | <ul><li>Breakpoints tool can be opened and tracks placed breakpoints</li><li>Breakpoints can be removed</li><li>Stack tool can be opened</li><li>LUA Locals tool can be opened</li></ul> |
```

Comment:
Request 1, 2, 4, & 5: Listify the methods

Reply:
Addressed in earlier point.